src/main/java/com/google/sps/data/TextAnalyser.java

```diff
@@ -183,13 +197,50 @@ private AnalyzeEntitiesResponse analyzeEntitiesText() throws IOException {
     return entities;
   }
 
+  public AnalyzeSyntaxResponse analyseSyntaxText() throws IOException {
```

Comment:
are you sure this exception is catched on the caller? seems like this is not the case.
If you are not sure if the caller catches it, you can always throw a custom exception (if you don't want to catch it yourself because the design prefers propagation).